contracts/bridge/BladeManager.sol

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.19;
+
+import "@openzeppelin/contracts-upgradeable/access/Ownable2StepUpgradeable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "../lib/GenesisLib.sol";
+import "../interfaces/bridge/IBladeManager.sol";
+import "../interfaces/bridge/IRootERC20Predicate.sol";
+
+contract BladeManager is IBladeManager, Ownable2StepUpgradeable {
+    using SafeERC20 for IERC20;
+    using GenesisLib for GenesisSet;
+
+    IRootERC20Predicate private _rootERC20Predicate;
+    GenesisSet private _genesis;
+
+    function initialize(address newRootERC20Predicate, GenesisAccount[] calldata genesisValidators) public initializer {
```

Comment:
```suggestion
    function initialize(address newRootERC20Predicate, GenesisAccount[] calldata genesisAccounts) public initializer {
```

Reply:
These accounts passed to `initialize` function will only be initial validators from L2, so I would leave this as it is, so that we are certain that these accounts are indeed validators.

Reply:
But, ok, we might want to initialize with non validator accounts as well. I'll rename it.